usaspending_api/etl/management/commands/copy_table_metadata.py

```diff
@@ -0,0 +1,146 @@
+        parser.add_argument(
+            "--skip-indexes",
+            action="store_true",
+            help="If provided, skips copying over the indexes to the destination table.",
+        )
+        parser.add_argument(
+            "--max-parallel-maintenance-workers",
+            type=int,
+            required=False,
+            help="Postgres session setting for max parallel workers for creating indexes."
+            " Only applicable if copy-indexes is provided.",
+        )
+        parser.add_argument(
+            "--maintenance-work-mem",
+            type=int,
+            required=False,
+            help="Postgres session setting for max memory to use for creating indexes (in GBs)."
+            "Only applicable if copy-indexes is provided.",
+        )
+        parser.add_argument(
+            "--index-concurrency",
+            type=int,
+            default=5,
+            help="Concurrency limit for index creation. Only applicable if copy-indexes is provided.",
+        )
+
+    def handle(self, *args, **options):
+        # Resolve Parameters
+        source_table = options["source_table"]
+        dest_table = options["dest_table"]
+        skip_constraints = options["skip_constraints"]
+        skip_indexes = options["skip_indexes"]
+        max_parallel_maintenance_workers = options["max_parallel_maintenance_workers"]
+        maintenance_work_mem = options["maintenance_work_mem"]
+        index_concurrency = options["index_concurrency"]
+
+        logger.info(f"Copying metadata from table {source_table} to table {dest_table}.")
+
+        copy_index_sql = None
+        with db.connection.cursor() as cursor:
+            if not skip_constraints:
+                logger.info(f"Copying constraints over from {source_table}")
+                copy_constraint_sql = make_copy_constraints(cursor, source_table, dest_table, drop_foreign_keys=True)
+                if copy_constraint_sql:
+                    cursor.execute("; ".join(copy_constraint_sql))
+                logger.info(f"Constraints from {source_table} copied over")
+
+            if not skip_indexes:
+                # Ensuring we're using the max cores available when generating indexes
+                if max_parallel_maintenance_workers:
+                    logger.info(f"Setting max_parallel_maintenance_workers to {max_parallel_maintenance_workers}")
+                    cursor.execute(f"SET max_parallel_maintenance_workers = {max_parallel_maintenance_workers}")
+                if maintenance_work_mem:
+                    logger.info(f"Setting maintenance_work_mem to '{maintenance_work_mem}GB'")
+                    cursor.execute(f"SET maintenance_work_mem = '{maintenance_work_mem}GB'")
+                copy_index_sql = make_copy_indexes(cursor, source_table, dest_table)
+        if copy_index_sql:
+            logger.info(f"Copying indexes over from {source_table}.")
+            CTSC_Command().create_indexes(copy_index_sql, index_concurrency)
```

Comment:
The `combine_transaction_search_chunks` command will eventually be removed. Can you move the `create_indexes` method to `copy_table_metadata` and have `combine_transaction_search_chunks` reference it from there instead?

Reply:
I'll give it a go. I'm a little worried it may run into a circular dependency.

Reply:
This looks good!